kpops/components/streams_bootstrap/__init__.py

```diff
@@ -29,3 +55,14 @@ class StreamsBootstrap(HelmApp, ABC):
         default=STREAMS_BOOTSTRAP_VERSION,
         description=describe_attr("version", __doc__),
     )
+
+    @pydantic.model_validator(mode="after")
+    def warning_for_latest_image_tag(self) -> Self:
+        if self.app.image_tag == "latest" and self.is_substituted():
```

Comment:
how about moving it as `field_validator` to `StreamsBootstrapValues`?

Reply:
If I move it under `StreamsBootstrapValues `, the component name cannot be fetched, and the log messages become very unclear.

Reply:
ok, that's sensible.
I would want to look into why it's not substituted though. Do you have any clues on that?

Reply:
I haven't quite found the why, but looking at the call stack, it shows a very twisted back-and-forth between the two classes `PipelineComponent` and `BaseDefaultsComponent`.

When we are applying a component here:
https://github.com/bakdata/kpops/blob/main/kpops/pipeline.py#L308
create an object of `PipelineComponent` that calls the init of its base class `BaseDefaultsComponent`. 

Then, in the init of the base class, we call
https://github.com/bakdata/kpops/blob/3d79e8edf151b1390b00c3162b40aa526a95a198/kpops/components/base_components/base_defaults_component.py#L88

Then, the init of the `PipelineCompnent` class is called again. This calls the super class's init method again when the `enrich` field is set to False. 

I assume this will initiate the child classes (along with the validators) before anything is enriched.